shennah_al-jawad/week_01/Fri-MTA.html

```diff
@@ -0,0 +1,205 @@
+// The L line has the following stops: 8th, 6th, Union Square, 3rd, and 1st
+// The 6 line has the following stops: Grand Central, 33rd, 28th, 23rd, Union Square, and Astor Place.
+// All 3 subway lines intersect at Union Square, but there are no other intersection points. (For example, this means the 28th stop on the N line is different than the 28th street stop on the 6 line, so you'll have to differentiate this when you name your stops in the arrays.)
+// Tell the user the number of stops AND the stops IN ORDER that they will pass through or change at.
+// Hints:
+
+// Work out how you would do it on paper first! Then start to explain that process in Javascript.
+// Get the program to work for a single line before trying to tackle multiple lines.
+// Don't worry about prompting the user for input. Hard code some values to get it working. You can use prompt() later to make it more interactive.
+// Consider diagramming the lines by sketching out the subway lines and their stops and intersection.
+// The key to the lab is finding the index positions of each stop. (hint: indexOf())
+// Make sure the stops that are the same for different lines have different names (i.e. 23rd on the N and on the 6 need to be differentiated)
+
+
+// Stations within Lines
+subway = {
+
+	lineN: ["Times Square", "34th", "28th N", "23rd N", "Union Square", "8th-N"],
+
+	lineL: ["8th L", "6th", "Union Square", "3rd", "1st"],
+
+	line6: ["Grand Central", "33rd", "28th 6", "23rd 6", "Union Square", "Astor Place"]
+}
+
+
+var planTrip = function(startLine, startStation, endLine, endStation) {
+
+	// total number of stops
+	var stops = 0;
+
+	// index of station in the array
+	var startingIndex;
+
+	// index of destination station
+	var destinationIndex;
+
+	var startingLine;
+
+	var destinationLine;
+
+	var stations = [];
+
+
+///////////////////// Start Line and Station //////////////////
+	// user input line N
+	if (startLine == "N") {
+		// variable startingLine is assigned Line N
+		startingLine = subway.lineN
+		console.log("the starting line has been set to N")
+		// looping through line N for starting station
+		for ( var i = 0; i < subway.lineN.length; i++ ) {
+
+			// checking users input against the current element in line N array
+			if (startStation === subway.lineN[i]) {
+				startingIndex = i;
+				console.log("the starting index has been set to " + i)
+			}
+		}
+	}
+
+	//user input line L
+	else if (startLine == "L") {
+		// variable startingLine is assigned Line L
+		startingLine = subway.lineL
+		console.log("the starting line has been set to L")
+		// looping through line L for starting station
+		for ( var i = 0; i < subway.lineL.length; i++ ) {
+
+			// checking users input against the current element in line L array
+			if (startStation === subway.lineL[i]) {
+				startingIndex = i;
+				console.log("the starting index has been set to " + i)
+			}
+		}
+	}
+
+	// user input line 6
+	else if (startLine == "6") {
```

Comment:
Did you notice that you repeating same code all the time, the only thing is changing is a name of the line, so to keep your code DRY (don't repeat yourself) you can create 1 function that set up line and pass in a station as a parameter. In this case you have 1 function but you can call on it multiply time.